File: selection_sort.py
def selection_sort(array):
  for i in range(len(array)):
    min_index = i
    for j in range(i+1, len(array)):
      if array[j] < array[min_index]:
        min_index = j

    array[i], array[min_index] = array[min_index], array[i]


arr = [5,4,7,2,15,16,11,1]
selection_sort(arr)
print("Sorted Array: ", arr)

File: selection_sort.js
function selectionSort(array){
    for (let i = 0; i < array.length; i++){
        min = i;
        for (let j = i+1; j < array.length; j++){
            if (array[j] < array[min]){
                min = j
            }
        }
        
        // swap the first unsorted element with the minimum element
        [array[i], array[min]] = [array[min], array[i]];
    }
    
    return array;
}

let arr = [56,23,6,2,0,9,1,0,32];
console.log("Sorted: ", selectionSort(arr));
